domain/application/state/application.go

```diff
@@ -1076,6 +1078,72 @@ func (st *ApplicationState) deleteSimpleUnitReferences(ctx context.Context, tx *
 	return nil
 }
 
+// GetSecretsForApplication returns the secrets owned by the specified application.
+func (st *ApplicationState) GetSecretsForApplication(
+	ctx domain.AtomicContext, appName string,
+) ([]*coresecrets.URI, error) {
```

Comment:
I find it strange that we return a slice of pointers to `coresecrets.URI`s. This seems unnecessary cognitive burden to place on the caller (i.e. should I check for nil, in this case, if there is no error, it is never nil).

Reply:
I'm not asking for anything here, just pointing it out.